CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+cmake_minimum_required(VERSION 3.10)
```

Comment:
I've seen recommendations to use at least 3.14 for "modern" features. I used 3.16 in #661.

Reply:
Right, on the flip side I don't know how onerous that is. 3.14 did only come out last year, I assume that for instance Ubuntu or Arch package managers for anything other than the newest OS versions might not have a sufficiently new CMake. Just want to make sure we're not causing problems for installation on e.g. any of our clusters like Comet.

Reply:
To be clear: you are absolutely right that I am using features that require 3.14 (all the `target_includes` that I'm doing don't exist in 3.10 AFAIK). I just want to make sure that won't be a problem for our HPC user base. HOOMD is still requiring only 3.9, although I don't know if that's just a legacy thing or planned to change for the v3 release.

Reply:
HOOMD v2.x builds with CMake 2.8.10.1 or newer. HOOMD v3.x will bump that requirement to 3.9. 3.9 was the first version to have a functioning findMPI that uses modern practices. I suggest supporting the oldest version you can without needing to implement excessive workarounds. This makes the software accessible to more users, many of whom do not update their software stacks that regularly.

Reply:
CMake version pi.0 (3.14.0) was [announced on Pi Day 2019](https://blog.kitware.com/cmake-3-14-0-available-for-download/). I'm fine with matching HOOMD's requirement of CMake 3.9 as long as it doesn't require significant workarounds. We provide conda-forge builds and wheels for freud, so far fewer users will build from source compared to HOOMD. Also since CMake is installable with pip or conda, I think it would be acceptable to have a version requirement 3.14 that is now >18 months old.

Reply:
With minor modifications we're able to support CMake >= 3.3. I modified CI to test against 3.6.3, which is the oldest version available on PyPI.